src/core/module_manager.cr

```diff
@@ -262,12 +262,19 @@ module PlaceOS
       if module_uri == uri
         driver = mod.driver.as(Model::Driver)
         driver_name = driver.name.as(String)
+        driver_id = driver.id.as(String)
         driver_file_name = driver.file_name.as(String)
         driver_commit = driver.commit.as(String)
 
         # Check if the module is on the current node
-        unless (driver_path = PlaceOS::Drivers::Compiler.is_built?(driver_file_name, driver_commit))
-          logger.tag_error("driver does not exist", driver_name: driver_name, driver_commit: driver_commit, module_id: mod_id)
+        unless (driver_path = PlaceOS::Drivers::Compiler.is_built?(driver_file_name, driver_commit, driver_id))
+          logger.tag_error(
+            message: "driver does not exist",
+            driver_name: driver_name,
+            driver_commit: driver_commit,
+            driver_id: driver_id,
+            module_id: mod_id,
+          )
```

Comment:
I think we should also attempt to compile the driver if we hit this?
If it's not already being compiled (assuming we can check for this) - or are we quite confident that we shouldn't end up here unless the driver is currently being compiled.

Just as if a user is triggering this attempting to get something working it would be good to automatically recover.
Alternatively maybe we periodically check if all the drivers are compiled (like every 10min) and trigger any new compilations that were somehow missed.

Reply:
we can't check that it's currently being compiled, but that would be a good addition to `Resource`

Reply:
i don't think attempting to compile here would be a good idea